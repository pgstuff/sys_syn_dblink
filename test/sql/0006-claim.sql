BEGIN;

CREATE EXTENSION sys_syn_dblink;

CREATE SCHEMA processor_data
        AUTHORIZATION postgres;

SELECT  dblink_connect('sys_syn_test', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
SELECT dblink_exec('sys_syn_test', 'BEGIN');

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('sys_syn_test', 'in');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('sys_syn_test', 'out');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('put');

SELECT sys_syn_dblink.processing_table_create (
        proc_schema     => 'processor_data',
        in_table_id     => 'test_table',
        out_group_id    => 'out',
        put_group_id    => 'put',
        dblink_connname => 'sys_syn_test');


SELECT * FROM processor_data.test_table_out_0_claim();


SELECT dblink_exec('sys_syn_test', 'ROLLBACK');
SELECT dblink_disconnect('sys_syn_test');

ROLLBACK;
