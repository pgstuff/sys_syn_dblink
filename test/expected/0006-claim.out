BEGIN;
CREATE EXTENSION sys_syn_dblink;
CREATE SCHEMA reader_data
        AUTHORIZATION postgres;
SELECT dblink_connect('sys_syn_test', 'dbname=contrib_regression');
 dblink_connect 
----------------
 OK
(1 row)

SELECT dblink_exec('sys_syn_test', 'BEGIN');
 dblink_exec 
-------------
 BEGIN
(1 row)

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('in');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('out');
SELECT sys_syn_dblink.reading_table_add (
        schema          => 'reader_data',
        in_table_id     => 'test_table',
        out_group_id    => 'out',
        in_group_id     => 'in',
        dblink_connname => 'sys_syn_test');
 reading_table_add 
-------------------
 
(1 row)

SELECT * FROM reader_data.test_table_out_claim();
 test_table_out_claim 
----------------------
 t
(1 row)

SELECT dblink_exec('sys_syn_test', 'ROLLBACK');
 dblink_exec 
-------------
 ROLLBACK
(1 row)

SELECT dblink_disconnect('sys_syn_test');
 dblink_disconnect 
-------------------
 OK
(1 row)

ROLLBACK;
